infra/gcp/ensure-staging-storage.sh

```diff
@@ -114,6 +120,7 @@ function ensure_staging_project() {
 
     # The names of the buckets
     local staging_bucket="gs://${project}" # used by humans
+    local staging_log_bucket="gs://${project}-log" # used by logs
```

Comment:
I am open to suggestion if people feel it's prudent to shard early and use something like `gs://k8s-infra-staging-gcslogs` instead.

Ultimately these logs need to live in `k8s-infra-public-pii`, and be accessible solely to contributors who are authorized to handle PII.  I figured using one bucket and one group for everything was simplest for now, until we find a need to add complexity.

We should also work to migrate the existing gcs logging to the automation and bucket described above, instead of being manually done and pointed at `k8s-artifacts-gcslogs`

Reply:
Thank you for a very clear explanation, I hope my pr addresses all of it.
- Set logging bucket constant in lib.sh
- Create function ensure_gcs_bucket_logging
- Call function from inside ensure_staging_project